web/src/features/time/TimeController.tsx

```diff
@@ -104,25 +104,31 @@ export default function TimeController({ className }: { className?: string }) {
     (timeRange: TimeRange) => {
       // Set time slider to latest value before switching aggregate to avoid flickering
       setSelectedDatetime({
-        datetime: selectedDatetime.datetime,
+        datetime: datetimes.at(-1),
         index: numberOfEntries,
       });
       setTimeRange(timeRange);
       trackEvent(TrackEvent.TIME_AGGREGATE_BUTTON_CLICKED, { timeRange });
     },
-    [setSelectedDatetime, selectedDatetime.datetime, numberOfEntries, setTimeRange]
+    [setSelectedDatetime, datetimes, numberOfEntries, setTimeRange]
```

Comment:
Note that `setTimeRange` was changing each render, see changes in its method body further below.